plugins/pastefromwordimage/plugin.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or http://ckeditor.com/license
+ */
+
+( function() {
+	'use strict';
+
+	CKEDITOR.plugins.add( 'pastefromwordimage', {
+		requires: 'pastefromword',
+		init: function() {}
+	} );
+
+	/**
+	 * Help methods used by paste from word image plugin.
+	 *
+	 * @since 4.8.0
+	 * @class CKEDITOR.plugins.pastefromwordimage
+	 */
+	CKEDITOR.plugins.pastefromwordimage = {
+		/**
+		 * Methods parse RTF content to find embedded images. Because Word shapes are also represnt in a smillar way as images,
+		 * method keeps track on them and return empty object in such case. That allows on having proper index of RTF images and image tags in html clipboard.
+		 *
+		 * @private
+		 * @since 4.8.0
+		 * @param {String} rtfContent Data obtained from RTF content.
+		 * @returns {Array} Contains array of objects with images or empty array if there weren't a match.
+		 * @returns {Object} return.Object Single image found in `rtfContent`.
+		 * @returns {String/null} return.Object.hex Hexadecimal string of image embedded in RTF clipboard.
+		 * @returns {String/null} return.Object.type String represent type of image, allowed values: 'image/png', 'image/jpeg' or `null`
+		 */
+		extractImagesFromRtf: function( rtfContent ) {
+			var ret = [],
+				rePictureHeader = CKEDITOR.env.mac ? /\{\\pict[\s\S]+?\\bliptag\-?\d+(\\blipupi\d+?)?\s?/ : /\{\\pict[\s\S]+?\{\\\*\\blipuid[\s0-9a-f]+?\}\s?/,
+				reShapeHeader = /\{\\shp[\s\S]+?\{\\\*\\svb\s?/,
+				rePictureOrShape = new RegExp( '(?:(' + rePictureHeader.source + ')|(' + reShapeHeader.source + '))([0-9a-f\\s]+)\\}', 'g' ),
+				unwantedType = CKEDITOR.env.mac ? '\\macpict' : '\\wmetafile',
```

Comment:
It would actually solve the need to use this hack: https://github.com/ckeditor/ckeditor-dev/pull/958/files#diff-086a650d277ef8af879ae71f10875ae1R62

Reply:
I've removed this logic but still hack needs to remain, because of this: https://github.com/ckeditor/ckeditor-dev/pull/958/files/9eab02612c88318ed129b7697d8e98c4ab584188#diff-d3d4bfb15d2eb9b8bcdfda799d8f109dR35
MacOS and Windows had 2 different rtf structures, which are recognised with 2 different patterns. So if we want to have working tests (for MacOS and Windows) running under single system, then env variable must be overwritten before test start.